modernize(tags): improve import tags api with detailed error reporting [#32972]

hassandotcms opened the pull request:
### Proposed Changes
  
  - improve api to return detailed error reporting
  - validate import statistics (totalRows, successCount, failureCount)
  - check error details with line numbers for failed rows
  - test both successful and partial import scenarios
  - add csv_with_errors.csv test file for error cases 
  
  resolves: #32972

This PR fixes: #32972

hassandotcms committed:
modernize(tags): improve import tags api with detailed error reporting

  - improve api to return detailed error reporting
  - validate import statistics (totalRows, successCount, failureCount)
  - check error details with line numbers for failed rows
  - test both successful and partial import scenarios
  - add csv_with_errors.csv test file for error cases
ref: #32972

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32972-modernize-import-tags-api

jdotcms commented on dotCMS/src/main/java/com/dotcms/rest/tag/TagsResourceHelper.java:
```diff
@@ -186,70 +204,115 @@ private void removeTempFolder(final File parentFolder) {
+     * @throws DotSecurityException If user lacks required permissions
      */
-    private File importTags(final File inputFile, final User user, final HttpServletRequest request)
+    private TagImportResult importTagsInternal(final File inputFile, final User user, final HttpServletRequest request)
```
Not sure if the temp api also have to be taken in consideration here

fabrizzio-dotCMS reviewed (approved):
just two minor nitpicks

hassandotcms committed:
modernize(tags): list tags api
  - Replace ImmutableList.copyOf() with List.copyOf() for modern Java standard
  - Change logging levels to DEBUG for consistency with existing INFO logs
  - Maintain header line counting consistency with content import behavior`

ref: #32972

hassandotcms committed:
fix postman tests.
ref: #32972

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32972-modernize-import-tags-api

hassandotcms commented on dotCMS/src/main/java/com/dotcms/rest/api/v2/tags/TagResource.java:
```diff
@@ -676,37 +680,79 @@ private InitDataObject getInitDataObject(final HttpServletRequest request,
+        return new ResponseEntityTagOperationView(
+            stats,
+            ImmutableList.copyOf(result.errors)
+    );
```
fixed

fmontes committed:
Merge branch 'main' into 32972-modernize-import-tags-api

hassandotcms committed:
Merge branch 'main' into 32972-modernize-import-tags-api

hassandotcms committed:
Merge branch 'main' of https://github.com/dotCMS/core into 32972-modernize-import-tags-api

hassandotcms committed:
modernize(tags): modernize import tags api (#32972)

minor refactoring, remove unused imports.

hassandotcms committed:
Merge branch 'main' into 32972-modernize-import-tags-api
